test/storage/cache_response.cpp

```diff
@@ -50,3 +50,175 @@ TEST_F(Storage, CacheResponse) {
 
     loop.run();
 }
+
+// Make sure we don't store a bad server response into the cache
+TEST_F(Storage, CacheNotFound) {
+    SCOPED_TEST(CacheNotFound);
+
+    using namespace mbgl;
+
+    util::RunLoop loop;
+    SQLiteCache cache(":memory:");
+    OnlineFileSource fs(&cache);
+
+    const Resource resource{ Resource::Unknown, "http://127.0.0.1:3000/not-found" };
+
+    // Insert existing data into the cache that will be marked as stale.
```

Comment:
Yeah, it's important because it verifies that it overwrites an existing "good" response (i.e. non-error response).